functions/src/sections/Section.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Section } from './Section.model';
+// TODO: replace once main repo is updated.
+import { UVicCourseScraper } from '@isaaccormack/uvic-course-scraper';
+import { db } from '../db/firestore';
+import { DetailedClassInformation } from '@isaaccormack/uvic-course-scraper/dist/src/types';
+
+export class SectionsService {
+  public async getSections(
+    term: string,
+    subject: string,
+    code: string
+  ): Promise<Section[]> {
+    const sections = await UVicCourseScraper.getCourseSections(
+      term,
+      subject,
+      code
+    );
+
+    return sections.map((e) => ({ crn: e.crn }));
+  }
+
+  public async getSectionSeats(
+    term: string,
+    subject: string,
+    code: string
+  ): Promise<DetailedClassInformation[]> {
+    console.log(term, subject, code);
+    // get term, subject, code to crn mappings from db.
+    const doc = await db.sectionMappings
```

Comment:
might be nice here to bring subject.toUpper() before checking the DB so subject='csc' and subject='CSC' both work.

Reply:
I'll add it at the controller level.